libraries/AP_BattMonitor/AP_BattMonitor_Logging.cpp

```diff
@@ -45,3 +49,106 @@ void AP_BattMonitor_Backend::Log_Write_BCL(const uint8_t instance, const uint64_
+
+    if (log_flag) {
+        struct log_BATI pkt{
+            LOG_PACKET_HEADER_INIT(LOG_BATI_MSG),
+            time_us             : AP_HAL::micros64(),
+            instance            : instance,
+            function            : MAV_BATTERY_FUNCTION_UNKNOWN,
+            type                : MAV_BATTERY_TYPE_UNKNOWN,
+            cells_in_series     : (uint8_t) (got_cells_in_series ? cells_in_series : 0),
+            capacity_design     : got_capacity_design ? capacity_design : -1,
+            capacity_full       : got_capacity_full ? capacity_full : -1,
+            design_voltage      : got_design_voltage ? design_voltage : -1,
+            cycles              : (uint16_t) (got_cycle_count ? cycles : UINT16_MAX),
+            serial_number       : {},
+            manufacture_date    : {},
+            product_name        : {},
+        };
+
+        strncpy_noterm(pkt.serial_number, serial_number, sizeof(pkt.serial_number));
+        strncpy_noterm(pkt.manufacture_date, manufacture_date, sizeof(manufacture_date));
+        strncpy_noterm(pkt.product_name, product_name, sizeof(pkt.product_name));
+        AP::logger().WriteBlock(&pkt, sizeof(pkt));
+        return;
+    }
+#endif // HAL_NO_LOGGING
+
+#ifndef HAL_NO_GCS
+    if (HAVE_PAYLOAD_SPACE(chan, SMART_BATTERY_INFO)) {
+
+        mavlink_msg_smart_battery_info_send(
+            chan,
+            instance,                       // Battery ID
+            MAV_BATTERY_FUNCTION_UNKNOWN,   // (not provided) Function of the battery
+            MAV_BATTERY_TYPE_UNKNOWN,       // (not provided) Type (chemistry) of the battery
+            got_capacity_design ? capacity_design : -1, // [mAh] Design capacity when full according to manufacturer, -1: field not provided.
+            got_capacity_full ? capacity_full : -1,     // [mAh] Capacity when full (accounting for battery degradation), -1: field not provided.
+            got_cycle_count ? cycles : UINT16_MAX,      // Charge/discharge cycle count. UINT16_MAX: field not provided.
+            serial_number,                  // Serial number in ASCII characters, 0 terminated. All 0: field not provided.
+            product_name,                   // Static device name. Encode as manufacturer and product names separated using an underscore.
+            0,                              // (not provided) [g] Battery weight. 0: field not provided.
+            UINT16_MAX,                     // (not provided) [mV] Minimum per-cell voltage when discharging. If not supplied set to UINT16_MAX value.
+            UINT16_MAX,                     // (not provided) [mV] Minimum per-cell voltage when charging. If not supplied set to UINT16_MAX value.
+            UINT16_MAX,                     // (not provided) [mV] Minimum per-cell voltage when resting. If not supplied set to UINT16_MAX value.
+            got_cells_in_series ? cells_in_series : 0,  // Number of battery cells in series. 0: field not provided.
```

Comment:
You already have the cells in series if you made it here, no need for the ternary.

Reply:
Yup removed.